frontend/src/utilities/useWatchBYONImages.ts

```diff
@@ -1,11 +1,15 @@
 import * as React from 'react';
-import { fetchBYONImages } from '~/services/imagesService';
 import { BYONImage } from '~/types';
-import useFetchState, { FetchState } from './useFetchState';
-import { POLL_INTERVAL } from './const';
+import { listImageStreams } from '~/api';
+import { mapImageStreamToBYONImage } from '~/utilities/imageStreamUtils';
+import useFetchState, { FetchState } from '~/utilities/useFetchState';
+import { POLL_INTERVAL } from '~/utilities/const';
 
-export const useWatchBYONImages = (): FetchState<BYONImage[]> => {
-  const getBYONImages = React.useCallback(() => fetchBYONImages(), []);
+export const useWatchBYONImages = (namespace: string): FetchState<BYONImage[]> => {
+  const getBYONImages = React.useCallback(async () => {
+    const imageStreams = await listImageStreams(namespace, 'byon');
```

Comment:
you should be able to use an actual watch now instead of polling. create a watchImageStreams ... i posted something in slack to see if this is actually allowed. its possible but may be deprecated so just FWI

Reply:
It should be good to use websockets for this. BFF isnt for a long time so no reason not to here

Reply:
lets just keep the same logic as to not complicate things

Reply:
@Gkrumbach07 changed to k8 watch